heartbeat/dovecot

```diff
@@ -0,0 +1,333 @@
+        return $OCF_SUCCESS
+    fi
+
+    # stop Dovecot
+    $binary $OPTIONS stop >/dev/null 2>&1
+    ret=$?
+
+    if [ $ret -ne 0 ]; then
+        ocf_exit_reason "Dovecot returned an error while stopping: " $ret
+        return $OCF_ERR_GENERIC
+    fi
+
+    # grant some time for shutdown and recheck 5 times
+    for i in 1 2 3 4 5; do
+        if dovecot_running info; then
+            sleep 1
+        else
+            break
+        fi
+    done
+
+    # dovecot stop did not succeed
+    if dovecot_running; then
+        ocf_exit_reason "Dovecot failed to stop."
+        return $OCF_ERR_GENERIC
+    fi
+
+    ocf_log info "Dovecot stopped."
+    return $OCF_SUCCESS
+}
+
+dovecot_reload()
+{
+    if dovecot_running; then
+        ocf_log info "Reloading Dovecot."
+        $binary $OPTIONS reload
+    fi
+}
+
+dovecot_monitor()
+{
+    local status_loglevel="err"
+
+    # Set loglevel to info during probe
+    if ocf_is_probe; then
+        status_loglevel="info"
+    fi
+
+    if dovecot_running $status_loglevel; then
+        return $OCF_SUCCESS
+    fi
+
+    return $OCF_NOT_RUNNING
+}
+
+dovecot_validate_all()
+{
+    # check that the Dovecot binaries exist and can be executed
+    check_binary "$binary"
+
+    # check config_file parameter
+    if [ "x$config_file" != "x" ]; then
+        if [ ! -f "$config_file" ]; then
+            if ocf_is_probe; then
+                ocf_log info "Dovecot configuration file '$config_file' not readable during probe."
+            else
+                ocf_exit_reason "Dovecot configuration file '$config_file' does not exist or is not readable."
+                return $OCF_ERR_INSTALLED
+            fi
+        fi
+    fi
+
+    return $OCF_SUCCESS
+}
+
+#
+# Main
+#
+
+if [ $# -ne 1 ]; then
+    usage
+    exit $OCF_ERR_ARGS
+fi
+
+binary=$OCF_RESKEY_binary
+config_file=$OCF_RESKEY_config_file
+parameters=$OCF_RESKEY_parameters
+
+
+# handle parameters
+case $1 in
+    meta-data)  meta_data
+                exit $OCF_SUCCESS
+                ;;
+
+    usage|help) usage
+                exit $OCF_SUCCESS
+                ;;
+esac
+
+# build Dovecot options string *outside* to access from each method
+OPTIONS=''
+OPTION_CONFIG_FILE=''
+
+# check if the Dovecot config_file exist
+if [ "x$config_dir" != "x" ]; then
+    # remove all trailing slashes
+    config_file=`echo $config_file | sed 's/\/*$//'`
+
+    # set OPTIONS if config_file is still set
+    # save OPTION_CONFIG_FILE seperatly
+    if [ "x$config_file" != "x" ]; then
+        OPTION_CONFIG_FILE="-c $config_file"
+        OPTIONS=$OPTION_CONFIG_FILE
+   fi
+fi
+
+# add all additional parameters to options string
+if [ "x$parameters" != "x" ]; then
+    OPTIONS="$OPTIONS $parameters"
+fi
+
+# check Dovecot status support
+status_support=false
+process_status=`$binary help 2> /dev/null | grep -q -e "process.*status"`
+ret=$?
+
+if [ $ret -eq 0 ]; then
+    status_support=true
+fi
+
+
+dovecot_validate_all
+ret=$?
+
+LSB_STATUS_STOPPED=3
```

Comment:
This doesn't seem to be used?

Reply:
Good catch, the postfix RA has the same issue.